observers.md

```diff
@@ -0,0 +1,87 @@
+
+The Observer is a central place outside of the original class to watch for database changes. By extracting that logic outside of the model, we bring back single responsiblity and avoid comingling models. Observers only report or take action on a process, and do not affect a process directly.
+
+An observer is an implementation of the [Observer pattern](https://en.wikipedia.org/wiki/Observer_pattern). You can [write your own](https://github.com/nslocum/design-patterns-in-ruby/tree/master/observer), or you can use Rails' [ActiveRecord::Observer](http://api.rubyonrails.org/v3.2/classes/ActiveRecord/Observer.html), which was removed from core in Rails 4.0 but is now [available as a gem](https://github.com/rails/rails-observers).
+
+In the example below, we're using a callback to initiate an action after creating a new user. We're mixing `MessageMailer` into the `User` class:
+
+```
+class User
+  after_create :notify
+
+  protected
+    def notify
+      MessageMailer.send_welcome_email(self).deliver
+    end
+end
+```
+
+As a better alternative, by using observers in the example below, we extract the `MessageMailer` logic out of `User` and put it in its own class, `UserObserver`, that watches for an event and dispatches an action accordingly:
+
+```
+class UserObserver < ActiveRecord::Observer
+  def after_create(user)
+    MessageMailer.send_welcome_email(user)
+  end
+end
+```
+
+Observers can also watch multiple classes:
+
+```
+class AuditObserver < ActiveRecord::Observer
+	observe :account, :balance
+
+	def after_update(record)
+		AuditTrail.new(record, "UPDATED")
+	end
+end
+```
+
+Obervers are especially helpful in cases like the above where there are multiple models comingled and we want to hook into all of them to watch for some event. We can pull logic out of models where it doesn't belong, put it in its own separate place, observe actions, and still affect change as needed.
+
+### Accounting for unseen side effects
+
+One thing to consider with observers is that we now have events that are not readily apparent in our observed class. Due to these unseen side effects, it can be less clear what's happening, and if there's a bug, it can be hard to track, since there is no direct link to the thing that is observing it.
+
+To mitigate this potential downside, we can write observers that reference the model they are observing. One way to do that is with `logger`. In this example from the [Rails documentation](http://api.rubyonrails.org/v2.3/classes/ActiveRecord/Observer.html), the observer uses `ActiveSupport::Logger` to log when specific callbacks are triggered:
```

Comment:
Ah, when I discussed the logger, what I meant was that the logger is an example of an observer that is seen. AR Observers are invisible in the sense that the object performing an action (like "create" or "save") does not have a visible reference to the observer. A developer looking at a User model that is watched by an observer can't see that the observer exists.

What do you think about naming this section "AR Observers and Invisibility" (or something) and discussing that downside?